src/js/components/ServicesBreadcrumb.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import {Link} from 'react-router';
+
+import ServicesPathUtil from '../utils/ServicesPathUtil';
+
+const breadcrumbIcon = (
+  <i className="icon icon-sprite icon-sprite-small icon-back forward" />
+);
+
+var ServicesBreadcrumb = React.createClass({
+  displayName: 'ServicesBreadcrumb',
+
+  propTypes: {
+    appId: React.PropTypes.string,
+    groupId: React.PropTypes.string,
+    taskId: React.PropTypes.string,
+    volumeId: React.PropTypes.string
+  },
+
+  getRootLink: function () {
+    const {groupId, appId} = this.props;
+
+    if (groupId == null ||
+      (groupId === '/' && appId == null)) {
+      return (
+        <span className="crumb">Services</span>
+      );
+    }
+
+    return (
+      <span className="crumb">
+        <Link to="services">Services</Link>
+      </span>
+    );
+  },
+
+  getGroupLinks: function () {
+    var {appId, groupId} = this.props;
+
+    if (groupId == null) {
+      return null;
+    }
+
+    const pathParts = groupId.split('/').slice(0, -1);
```

Comment:
I may hook in?

If performance is a concern then @pierlo-upitup approach is the faster one I guess, because there is no need to filter the tree.
Each filter operation has a cost of 3 function calls (filterItemsById, startsWith, getId) for each step in the iteration.
I mean it's not too bad, but the string fiddeling approach is much more straight forward in this case.

On the other hand I find @orlandohohmeier approach more descriptive.

But I tend to the performance approach here, that is descriptive too..

Just my 2¢ and only a humble opinion.

Don't let us waist time on discussing on elegance.
"Beauty is in the eye of the beholder."